File: code/geometry_py/proj_pl.py
def projPL(a, b, p, refl = False):
	v = vecsub(b, a)
	s = (1 + refl) * cross(b, p, a) / len2(v)
	return (p[0] + v[1] * s, p[1] - v[0] * s)

File: code/geometry_cpp/proj_pl.cpp
dpoint projPL(ipoint a, ipoint b, ipoint p, bool refl = false) {
	auto v = b - a;
	double s = (1 + refl) * b.cross(p, a) / v.len2();
	return { p.x + v.y * s, p.y - v.x * s };
}
